platform_plugin_aspects/signals.py

```diff
@@ -63,34 +64,33 @@ def receive_course_enrollment_changed(  # pylint: disable=unused-argument  # pra
     )
 
 
-def on_user_profile_updated(instance):
-    """
-    Queues the UserProfile dump job when the parent transaction is committed.
-    """
-    # import here, because signal is registered at startup, but items in tasks are not yet able to be loaded
-    from platform_plugin_aspects.tasks import (  # pylint: disable=import-outside-toplevel
-        dump_data_to_clickhouse,
-    )
-
-    sink = UserProfileSink(None, None)
-    dump_data_to_clickhouse.delay(
-        sink_module=sink.__module__,
-        sink_name=sink.__class__.__name__,
-        object_id=str(instance.id),
-    )
-
-
 def on_user_profile_updated_txn(**kwargs):
     """
     Handle user_profile saves in the middle of a transaction.
 
+    Handle saves in the middle of a transaction.
     If this gets fired before the transaction commits, the task may try to
     query an id that doesn't exist yet and throw an error. This should postpone
     queuing the Celery task until after the transaction is committed.
     """
-    transaction.on_commit(
-        lambda: on_user_profile_updated(kwargs["instance"])
-    )  # pragma: no cover
+
+    def on_user_profile_updated(instance):
```

Comment:
Looks like there is an issue here:

```
2024-10-01 15:11:04,575 ERROR 1 [django.request] [user 6] [ip 192.168.247.1] log.py:241 - Internal Server Error: /api/user/v1/accounts/bmtcril
Traceback (most recent call last):
  File "/openedx/venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/openedx/venv/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/pyenv/versions/3.11.8/lib/python3.11/contextlib.py", line 80, in inner
    with self._recreate_cm():
  File "/openedx/venv/lib/python3.11/site-packages/django/db/transaction.py", line 307, in __exit__
    connection.set_autocommit(True)
  File "/openedx/venv/lib/python3.11/site-packages/django/db/backends/base/base.py", line 501, in set_autocommit
    self.run_and_clear_commit_hooks()
  File "/openedx/venv/lib/python3.11/site-packages/django/db/backends/base/base.py", line 779, in run_and_clear_commit_hooks
    func()
  File "/mnt/platform-plugin-aspects/platform_plugin_aspects/signals.py", line 93, in <lambda>
    transaction.on_commit(lambda: on_user_profile_updated(**kwargs))
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: on_user_profile_updated_txn.<locals>.on_user_profile_updated() got an unexpected keyword argument 'signal'
```

Changing it to `def on_user_profile_updated(sender, instance, **kwargs):` seemed to fix it

Reply:
Updated. The taxonomy saved workflow can be tested from the django admin